index.js

```diff
@@ -9,29 +9,26 @@ function ToObject(val) {
 	return Object(val);
 }
 
-function ownEnumerableKeys(obj) {
-	var keys = Object.getOwnPropertyNames(obj);
-
-	if (Object.getOwnPropertySymbols) {
-		keys = keys.concat(Object.getOwnPropertySymbols(obj));
-	}
-
-	return keys.filter(function (key) {
-		return propIsEnumerable.call(obj, key);
-	});
-}
-
 module.exports = Object.assign || function (target, source) {
 	var from;
 	var keys;
 	var to = ToObject(target);
+	var symbols;
 
 	for (var s = 1; s < arguments.length; s++) {
-		from = arguments[s];
-		keys = ownEnumerableKeys(Object(from));
+		from = Object(arguments[s]);
+
+		for (var key in from) {
+			if (hasOwnProperty.call(from, key)) {
+				to[key] = from[key];
+			}
+		}
 
-		for (var i = 0; i < keys.length; i++) {
-			to[keys[i]] = from[keys[i]];
+		if (Object.getOwnPropertySymbols) {
```

Comment:
Yep, this is described by [9.1.12](http://www.ecma-international.org/ecma-262/6.0/#sec-ordinary-object-internal-methods-and-internal-slots-ownpropertykeys) in spec:

```
When the [[OwnPropertyKeys]] internal method of O is called the following steps are taken:
    1. Let keys be a new empty List.
    2. For each own property key P of O that is an integer index, 
in ascending numeric index order add P as the last element of keys.
    3. For each own property key P of O that is a String but is not an integer index, 
in property creation order add P as the last element of keys.
    4. For each own property key P of O that is a Symbol, in property creation order add P as the last element of keys.

Return keys.
```